api/routes.py

```diff
@@ -190,10 +190,15 @@ async def get_closest_mappings_for_text(text: str,
                                         terminology_name: str = "SNOMED CT",
                                         model: str = "sentence-transformers/all-mpnet-base-v2",
                                         limit: int = 5):
-    embedding_model = MPNetAdapter(model)
+    if model == "text-embedding-ada-002":
+        embedding_model = GPT4Adapter(model)
+    elif model == "sentence-transformers/all-mpnet-base-v2":
+        embedding_model = MPNetAdapter(model)
+    else:
+        raise HTTPException(status_code=400, detail="Unsupported embedding model.")
```

Comment:
This can be handled from the exception from the SentenceTransformer class

Reply:
I don't see how it handles this, could you point me to the line